src/partition.rs

```diff
@@ -226,8 +226,52 @@ impl BlockSize {
     }
   }
 
+  /// Source: Subsampled_Size (AV1 specification section 5.11.38)
+  pub fn subsampled_size(self, xdec: usize, ydec: usize) -> BlockSize {
+    match (xdec, ydec) {
+      (0, 0) /* 4:4:4 */ => self,
+      (1, 0) /* 4:2:2 */ => match self {
+        BLOCK_4X4 | BLOCK_8X4 => BLOCK_4X4,
```

Comment:
Right, this is rustic way!
The whole ss_size_lookup[ ] table and how ss_size is lookup-ed has been hastly copied from libaom.